src/actions/promote.ts

```diff
@@ -74,6 +75,10 @@ async function promoteBlueGreen(kubectl: Kubectl) {
 
     // checking stability of newly created deployments 
     const deployedManifestFiles = result.newFilePaths;
+
+    // Adding traceability data
+    await AzureTraceabilityHelper.addTraceability(deployedManifestFiles);

```

Comment:
There are some calls to `adjust/route traffic` (in `promote/reject` flows for `canary` and `blue-green`) that are technically modifying the cluster as a part of this deployment. Although the objects of higher interest are the ones related to workloads and they get captured with the current approach.

Perhaps, the ideal hook for capturing traceability seems to be `kubectl.apply` (something like - have a property attached with `kubectl` object to track all deployed manifests. This keeps updating on each call to `apply`. Flush this towards the end of the action execution). Thoughts?

That said, we are currently living with _not capturing_ these objects with `annotations` approach. Maybe it makes sense to continue with that approach.

Reply:
Moved stuff to kubectl apply